src/main/java/com/google/sps/data/DatastoreManager.java

```diff
@@ -34,6 +36,10 @@
      * @return ID of entity inserted into datastore
      */ 
     public static long insertCommentInDatastore(String name, String message) {
+        ToxicityDetector detector = new ToxicityDetector(); 
+        if (detector.isCommentToxic(message)) {
+            throw new IllegalArgumentException("Can't enter toxic comments."); 
```

Comment:
I wouldn't use an IllegalArgumentException. For two main reasons:
1. Its stretching the definition of the exception, which is generally used for obviously wrong input. Say an empty string or null object, possibly indicating a bug somewhere else in the code. Whether a comment is toxic or not isn't obvious and could happen all the time.
2. Its a runtime exception (i.e unchecked exception) which is meant for unexpected behavior. 

Here toxic comments are expected to happen and you want any caller of this method to know and handle that accordingly. So you want a checked exception that enforces that. You can create your own exception type such as ToxicCommentException or more generically a DatastoreException that you could reuse across this class.

Happy to explain more.

Reply:
+1 please address this.